frappe/contacts/doctype/address/address.py

```diff
@@ -53,6 +54,18 @@ def link_address(self):
 
 		return False
 
+	def set_link_title(self):
+		if not self.links:
+			return
+		else:
+			for address in self.links:
+				if not address.link_title:
+					linked_doc = frappe.get_doc(address.link_doctype, address.link_name)
+					try:
+						address.link_title = linked_doc.title_field
```

Comment:
I think `linked_doc.get("title_field") or linked_doc.get("name")` would be more readable.